src/main/java/com/hms/flexyosisoftconnector/OSIsoftConnectorMain.java

```diff
@@ -170,78 +179,105 @@ private static void getNewPayloadToSend() {
         isMemoryCurrentlyLow = false;
       }
 
-      // Process datapoints from queue
-      Logger.LOG_DEBUG("Getting next section of datapoints");
-      ArrayList queuePoints = new ArrayList();
-      try {
-        if (!HistoricalDataQueueManager.doesTimeTrackerExist()) {
-          // Start a new time tracker file at current time
-          HistoricalDataQueueManager.getFifoNextSpanDataAllGroups(true);
-        } else {
-          // Get the data points for current time period in time tracker
-          queuePoints = HistoricalDataQueueManager.getFifoNextSpanDataAllGroups(false);
-        }
-      } catch (IOException e) {
-        Logger.LOG_EXCEPTION(e);
-        Logger.LOG_WARN(
-            "Exception thrown when obtaining new batch of datapoints. The queue will retry.");
-      } catch (CircularizedFileException e) {
-        Logger.LOG_EXCEPTION(e);
-        Logger.LOG_WARN("Data in the historical log has been overwritten with newer data.");
-      } catch (EbdTimeoutException e) {
-        Logger.LOG_EXCEPTION(e);
-        Logger.LOG_WARN("EBD call has timed out while fetching data.");
-      } catch (JSONException e) {
-        Logger.LOG_EXCEPTION(e);
-        Logger.LOG_WARN("Invalid JSON received while fetching data.");
-      } catch (TimeTrackerUnrecoverableException e) {
-        Logger.LOG_EXCEPTION(e);
-        Logger.LOG_WARN("Time tracker file is not recoverable.");
-      } catch (CorruptedTimeTrackerException e) {
-        Logger.LOG_EXCEPTION(e);
-        Logger.LOG_WARN("Time tracker file has been corrupted.");
-      } catch (Exception e) {
-        Logger.LOG_EXCEPTION(e);
-        Logger.LOG_WARN("There was an unknown error collecting data points!");
-      }
+      // If the payload queue is not full, read from historical data
+      if (!payloadMngr.isQueueFull()) {
+
+        payloadQueueFullCount = 0;
+        payloadQueueFullWarnThresh = 1;
 
-      // if there are data points to send
-      if (queuePoints.size() != 0) {
-        // Check if queue is behind
+        // Process datapoints from queue
+        Logger.LOG_DEBUG("Getting next section of datapoints");
+        ArrayList queuePoints = new ArrayList();
         try {
-          long queueBehindMillis = HistoricalDataQueueManager.getQueueTimeBehindMillis();
-          if (queueBehindMillis >= OSIsoftConfig.WARNING_LIMIT_QUEUE_BEHIND_MILLISECONDS) {
-            Logger.LOG_WARN(
-                "The historical data queue is running behind by "
-                    + SCTimeUtils.getDayHourMinSecsForMillis(queueBehindMillis));
+          if (!HistoricalDataQueueManager.doesTimeTrackerExist()) {
+            // Start a new time tracker file at current time
+            HistoricalDataQueueManager.getFifoNextSpanDataAllGroups(true);

```

Comment:
please remind me why magic booleans are bad?

Reply:
The presence of magic booleans can make the code much more difficult to read or understand. 

Reading the following does not explain or hint at what the `true` means or does and could leave one questioning:
```
            HistoricalDataQueueManager.getFifoNextSpanDataAllGroups(true);
```

Reading the following does explain what the `true` means/does and is more friendly:
```
            final boolean startNewTimeTracker = true;
            HistoricalDataQueueManager.getFifoNextSpanDataAllGroups(startNewTimeTracker);
```

Reply:
It looks like an unnecessary initialization that needlessly get's called hundreds of thousands of times.  Don't we all have nice IDE's that do this for us. 
![image](https://user-images.githubusercontent.com/72526279/236929622-e56d238f-9920-47d5-ac60-11c97bdfdcfe.png)

Reply:
It may appear unnecessary, but it greatly affects code readability and easy understanding.

A nice Java IDE such as IntelliJ or VS Code may/should include parameter naming hints like those shown above, but there is no guarantee that everyone will have an IDE that supports this. Additionally, viewing on GitHub does not include these parameter hints.